test/release/examples/benchmarks/shootout/fasta.chpl

```diff
@@ -96,47 +101,93 @@ proc repeatMake(desc, alu, n) {
 }
 
 //
-// Output a random sequence of length 'n' using distribution a
+// Output a random sequence of length 'n' using distribution 'a'
 //
-proc randomMake(desc, nucleotides, n) {
+proc randomMake(desc, nuclInfo, n) {
   stdout.writeln(desc);
 
-  const numNucls = nucleotides.size;
-  var cumulProb: [1..numNucls] int;
+  const numNucls = nuclInfo.size;
+  var cumulProb: [1..numNucls] randType;
+
+  // compute the cumulative probabilities of the nucleotides
   var p = 0.0;
   for i in 1..numNucls {
-    p += nucleotides[i](prob);
-    cumulProb[i] = 1 + (p*IM):int;
+    p += nuclInfo[i](prob);
+    cumulProb[i] = 1 + (p*IM):randType;
   }
 
-  var line_buff: [0..lineLength] int(8);
+  // guard when tasks can access the random numbers or output stream
+  var randGo, outGo: [0..#numTasks] atomic int;
+  randGo.write(1);
+  outGo.write(1);
+
+  coforall tid in 0..#numTasks {
+    const chunkSize = lineLength*blockSize,
+          nextTask = (tid + 1) % numTasks;
+
+    var myBuff: [0..(lineLength+1)*blockSize-1] int(8),
+        myRands: [0..chunkSize] int/*(32)*/;
+
+    for i in 1..n by chunkSize*numTasks align 1+tid*chunkSize {
```

Comment:
I was mentally wrestling with this loop yesterday as well.  What I like about it is that it essentially says "We're going from 1..n" what I don't like about it is that the pattern is so carefully orchestrated but unclear.  I found myself wishing for an iterator from the standard modules that would help me with this.  Something like 'for i in roundRobin(1..n, chunkSize, tid, numTasks).'  This relates a bit to the RangeChunking library yet is different (I believe).  That said, it's also intertwined with the semi-magic numbers that are stored into the atomics to pass the token which seems similarly unfortunate...

All that said, without a library routine, the other way I was thinking of writing it yesterday was:

    for 1+tid*chunkSize..n  by chunkSize*numTasks

if you think that's any better.  I'm not sure it is so maybe the comment is the way to go for now.